### PR TITLE
[WIP] VACMS-3969: Install focal point, remove imagewidgetcrop and update me…

### PR DESCRIPTION
**Pull request by msbtterswrth**
…dia crop options.

## Description

Relates to #3969. 

## Testing done
manual, automatic

## Screenshots
Node edit: 
image

Image preview: 
image

Node view:
image
image

## QA steps

What needs to be checked to prove this works?  
What needs to be checked to prove it didn't break any related things?  
What variations of circumstances (users, actions, values) need to be checked?  

As a cms editor
1. Create an Event (`/node/event/add`)
   - [ ] Validate that the event photo field is now an inline entity form
   - [ ] Name and section of media are automatically populated on node save
2. Edit a Vet Center (`/node/3862/edit`)
   - [ ] Validate that the facility photo field is now an inline entity form
   - [ ] Name and section of media are automatically populated on node save
3. Then do this for
   - [ ] Event (`/node/46208/edit`)
   - [ ] Story (`/node/46110/edit`)
   - [ ] VAMC System (`/node/15018/edit`)
   - [ ] VAMC Facility (`/node/37668/edit`)
   - [ ] Vet Center CAP (`/node/44863/edit`)
   - [ ] Mobile Vet Center (`node/17464/edit`)
   - [ ] Vet Center Outstation (`/node/17536/edit`)
4. Then build the frontend
   - [ ] Validate selected focal point saves and is properly displayed on the frontend for each content type
     - [ ] Event (https://web-uil7px8rg8m54linmybwia3hm1qktvdm.ci.cms.va.gov/tomah-health-care/events/46127)
     - [ ] Story (https://web-uil7px8rg8m54linmybwia3hm1qktvdm.ci.cms.va.gov/pacific-islands-health-care/stories/directors-message-april-29-2022)
     - [ ] VAMC System (https://web-uil7px8rg8m54linmybwia3hm1qktvdm.ci.cms.va.gov/poplar-bluff-health-care)
     - [ ] VAMC Facility (https://web-uil7px8rg8m54linmybwia3hm1qktvdm.ci.cms.va.gov/tomah-health-care/locations/wausau-va-clinic)
     - [ ] Vet Center (https://web-uil7px8rg8m54linmybwia3hm1qktvdm.ci.cms.va.gov/jackson-vet-center) 
     - [ ] Vet Center CAP (`/node/44863/edit`)
     - [ ] Mobile Vet Center (`node/17464/edit`)
     - [ ] Vet Center Outstation (`/node/17536/edit`)
5. Then validate Acceptance Criteria from issue
   - [ ] Inline Entity Form-Complex widget is applied to applicable media field(s)
   - [ ] Name" and "description" sections are hidden
   - [ ] image widget crop is disabled
   - [ ] focalpoint module is installed and enabled
   - [ ] Handoff for UX writing updates and design review

### Definition of Done

- [ ] Documentation has been updated, if applicable.
- [ ] Tests have been added if necessary.
- [ ] Automated tests have passed.
- [ ] Code Quality Tests have passed.
- [ ] Acceptance Criteria in related issue are met.
- [ ] Manual Code Review Approved.

### Select Team for PR review

- [ ] `Platform CMS Team`
- [ ] `Sitewide program`
- [x] `⭐️ Sitewide CMS`
- [ ] `⭐️ Public websites`
- [ ] `⭐️ Facilities`
- [ ] `⭐️ User support`

### Is this PR blocked by another PR?

- [ ] `DO NOT MERGE`

### Does this PR need review from a Product Owner

- [ ] `Needs PO review`

### CMS user-facing annoucement

Is an announcement needed to let editors know of this change?
- [ ] Yes, and it's written in issue ____ and queued for publication.
  - [ ] Merge and ping @ rachel-kauff so she's ready to publish after deployment
- [x] Yes, but it hasn't yet been written
  - [ ] Don't merge yet -- ping @ rachel-kauff to prompt her to write and queue content
- [ ] No announcement is needed for this code change.
  - [ ] Merge & carry on unburdened by announcements

### CODE REVIEW

**Comment by msbtterswrth** [comment]
Folded this into #9058 so i think we can probably close this one.

**Comment by rachel-kauff** [comment]
I think if we end up using focal point, we'll need to see about some design affordances that make that crosshair thing more usable.

**Comment by msbtterswrth** [comment]
Took a stab at improving that pointer UX. The pointer in the FA icon points to the exact focal point. 
image

File: docroot/modules/custom/va_gov_media/va_gov_media.module [hunk 1]
```diff
@@ -48,6 +48,21 @@ function va_gov_media_entity_update(EntityInterface $entity) {
   _va_gov_media_image_style_warmer_warm_up($entity);
 }
 
+/**
+ * Implements hook_entity_presave().
+ */
+function va_gov_media_entity_presave(EntityInterface $entity) {
```
Comment (swirtSJW) [inline docroot/modules/custom/va_gov_media/va_gov_media.module:54]: Big picture we are trying to move all these kinds of hooks to event subscribers.   We still have a bunch laying around in code that need to be converted, but Ideally I would like to not add new ones.

File: docroot/modules/custom/va_gov_media/va_gov_media.module [hunk 2]
```diff
@@ -72,3 +87,18 @@ function _va_gov_media_image_style_warmer_warm_up(EntityInterface $entity) {
...
   }
 }
+
+/**
+ * Autofill some media fields so that they can be hidden for editor UX.
+ *
+ * @param \Drupal\Core\Entity\FieldableEntityInterface $entity
+ *   Entity.
+ */
+function _va_gov_media_library_ux_helper(FieldableEntityInterface $entity) {
```
Comment (swirtSJW) [inline docroot/modules/custom/va_gov_media/va_gov_media.module:97]: This function does only one thing, which is great, that is what a function is supposed to do. 
However, it has been named as though it is not sure what it does.  It does not appear to do anything related to a library and 'helper' doesn't tell us anything. 

Whenever we're naming something generic like helper, processor, doer, fixer it should set off your spidey sense that either
a) the function is trying to do too much and can't be named well (means it should probably be broken down to simpler functions, or
b) that it needs a more specific name.

Whenever possible, name functions that are as specific as possible about what they do.  It makes the code a lot more readable and as such more maintainable.

Something like `_va_gov_media_set_media_owner` would be good
and `_va_gov_media_set_media_owner_to_node_owner` is even better.  It tells exactly what it does.

File: docroot/modules/custom/va_gov_media/va_gov_media.module [hunk 3]
```diff
@@ -72,3 +87,18 @@ function _va_gov_media_image_style_warmer_warm_up(EntityInterface $entity) {
...
+  $section = $entity->get('field_administration')->target_id;
+  $media_entity = $entity->get('field_media')->entity;
+  if (!is_null($media_entity)) {
+    $media_entity->set('field_owner', ['target_id' => $section]);
+    $media_entity->save();
+  }
```
[the remark this answers is not in the document]
Comment (swirtSJW) [reply docroot/modules/custom/va_gov_media/va_gov_media.module:103]: I think the problem that you are running into with this is that you are saving the media item here in the presave and then it gets saved again during node save,  which is likely overwriting your new value with an empty value making it seem like this hook never fired.

**Comment by msbtterswrth** [comment]
@swirtSJW Just pushed up a JS approach. Figured out one thing that might throw a tiny wrench into things: when IEF has a required field it basically makes the optional parent field required, there seems to be a patch [here](https://www.drupal.org/project/inline_entity_form/issues/3099844#comment-13907770) that might work for us, or i'm sure we can solve this.

**Comment by ndouglas** [comment]
I can confirm that neither [patch #15](https://www.drupal.org/files/issues/2020-11-18/inline_entity_form-required_fields_optional_ief-3099844-15.patch) nor [patch #24](https://www.drupal.org/files/issues/2021-07-19/inline_entity_form-required_fields_optional_ief-3099844-24.patch) solve the issue with the image field showing up as required at e.g. `node/add/event`.

**Comment by timcosgrove** [comment]
Old, not moving forward. Closing.

**Pull request closed without merging**